rust/gui-client/src-tauri/src/client/gui/system_tray.rs

```diff
@@ -105,172 +165,325 @@ impl Icon {
+                    .copyable("Internet")
+                    .copyable("")
+                    .separator()
+                    .disabled("Site")
+                    .copyable("test")
+                    .copyable(ALL_GATEWAYS_OFFLINE),
+            )
+            .add_bottom_section(DISCONNECT_AND_QUIT); // Skip testing the bottom section, it's simple
+
+        assert_eq!(
+            actual,
+            expected,
+            "{}",
+            serde_json::to_string_pretty(&actual).unwrap(),
+        );
     }
 
-    fn item<E: Into<Option<Event>>, S: Into<String>>(self, id: E, title: S) -> Self {
-        self.add_item(item(id, title))
+    #[test]
+    fn some_resources_one_favorite() {
+        let resources = resources();
+        let input = AppState::SignedIn(SignedIn {
+            actor_name: "Jane Doe",
+            resources: &resources,
+        });
+        let actual = input.into_menu();
+        let expected = Menu::default()
+            .disabled("Signed in as Jane Doe")
+            .item(Event::SignOut, SIGN_OUT)
+            .separator()
+            .disabled(RESOURCES)
+            .add_submenu(
+                "172.172.0.0/16",
+                Menu::default()
+                    .copyable("cidr resource")
+                    .separator()
+                    .disabled("Resource")
+                    .copyable("172.172.0.0/16")
+                    .copyable("172.172.0.0/16")
+                    .separator()
+                    .disabled("Site")
+                    .copyable("test")
+                    .copyable(NO_ACTIVITY),
+            )
+            .add_submenu(
+                "gitlab.mycorp.com",
+                Menu::default()
+                    .copyable("dns resource")
+                    .separator()
+                    .disabled("Resource")
+                    .copyable("gitlab.mycorp.com")
+                    .copyable("gitlab.mycorp.com")
+                    .separator()
+                    .disabled("Site")
+                    .copyable("test")
+                    .copyable(GATEWAY_CONNECTED),
+            )
+            .add_submenu(
+                "Internet",
+                Menu::default()
+                    .copyable("")
+                    .separator()
+                    .disabled("Resource")
+                    .copyable("Internet")
+                    .copyable("")
+                    .separator()
+                    .disabled("Site")
+                    .copyable("test")
+                    .copyable(ALL_GATEWAYS_OFFLINE),
+            )
+            .add_bottom_section(DISCONNECT_AND_QUIT); // Skip testing the bottom section, it's simple
+
+        assert_eq!(
+            actual,
+            expected,
+            "{}",
+            serde_json::to_string_pretty(&actual).unwrap()
+        );
     }
 
-    fn separator(self) -> Self {
-        self.add_native_item(SystemTrayMenuItem::Separator)
+    #[test]
+    fn some_resources_invalid_favorite() {
```

Comment:
Nit: Technically, this PR doesn't know about favorites.

Reply:
Yeah. I'll consolidate those tests. It'll make the diff a little odd for the favorites PR but that's fine